apache-kafka-3/src/test/java/com/baeldung/kafka/KafkaProducerRetriesLiveTest.java

```diff
@@ -0,0 +1,99 @@
+package com.baeldung.kafka;
+
+import static java.util.Collections.singleton;
+import static org.apache.kafka.clients.producer.ProducerConfig.BOOTSTRAP_SERVERS_CONFIG;
+import static org.apache.kafka.clients.producer.ProducerConfig.DELIVERY_TIMEOUT_MS_CONFIG;
+import static org.apache.kafka.clients.producer.ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG;
+import static org.apache.kafka.clients.producer.ProducerConfig.REQUEST_TIMEOUT_MS_CONFIG;
+import static org.apache.kafka.clients.producer.ProducerConfig.RETRIES_CONFIG;
+import static org.apache.kafka.clients.producer.ProducerConfig.RETRY_BACKOFF_MS_CONFIG;
+import static org.apache.kafka.clients.producer.ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.ExecutionException;
+
+import org.apache.kafka.clients.admin.AdminClient;
+import org.apache.kafka.clients.admin.NewTopic;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.serialization.StringSerializer;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.testcontainers.containers.KafkaContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.utility.DockerImageName;
+
+@Testcontainers
+class KafkaProducerRetriesLiveTest {
+
+    static AdminClient adminClient;
+    
+    @Container
+    static final KafkaContainer KAFKA_CONTAINER = new KafkaContainer(
+      DockerImageName.parse("confluentinc/cp-kafka:latest"));
+
+    @BeforeAll
+    static void beforeAll() {
+        Properties props = new Properties();
+        props.put("bootstrap.servers", KAFKA_CONTAINER.getBootstrapServers());
+        adminClient = AdminClient.create(props);
+    }
+
+    @Test
+    void whenMessageCannotBeSent_thenKafkaProducerRetries_usingDefaultConfig() throws Exception {
+        // given
```

Comment:
let's remove the comments (_given_ _when_ and _then_ are not necessary)

Reply:
I have removed the comments and re-formatted the file now 👍